frontend/src/metabase/embedding-sdk/types/components/data-picker.ts

```diff
@@ -2,10 +2,11 @@ import type { TableId } from "metabase-types/api";
 
 export interface DataSourceSelectorProps {
   isInitiallyOpen: boolean;
+  /** false when joining data, true otherwise */
   canChangeDatabase: boolean;
   selectedDatabaseId: number | null;
-  selectedTableId: TableId | undefined;
-  selectedCollectionId: number | null | undefined;
+  selectedTableId?: TableId;
+  selectedCollectionId?: number | null;
```

Comment:
make them optional, they already included `undefined` previously, but doing this will make unit testing easier.